src/pgrepr/src/value.rs

```diff
@@ -144,12 +184,26 @@ impl Value {
             Value::Numeric(n) => strconv::format_decimal(buf, &n.0),
             Value::Text(s) => buf.put(s.as_bytes()),
             Value::Jsonb(js) => strconv::format_jsonb(buf, js),
+            Value::List(elems) => {
+                buf.put(&b"{"[..]);
+                let mut elems = elems.iter().peekable();
+                while let Some(elem) = elems.next() {
+                    match elem {
+                        Some(elem) => elem.encode_text(buf),
+                        None => buf.put(&b"NULL"[..]),
+                    }
+                    if elems.peek().is_some() {
+                        buf.put(&b","[..])
+                    }
+                }
+                buf.put(&b"}"[..]);
+            }
         }
     }
 
     /// Serializes this value to `buf` using the [binary encoding
     /// format](Format::Binary).
-    pub fn encode_binary(&self, buf: &mut BytesMut) {
+    pub fn encode_binary(&self, buf: &mut BytesMut) -> Result<(), Box<dyn Error + Sync + Send>> {
```

Comment:
Shoot, it was intentional that text/binary encoding was infallible. Makes the API much nicer, and the invariant that encoding cannot fail is pretty reasonable IMO. We should invent our own binary encoding for `Value::List`, and for now the binary encoding can just be defined to be "the same as the text encoding."

Reply:
> for now the binary encoding can just be defined to be "the same as the text encoding."

That might be surprising? But I guess it at least leaves the door open for a proper encoding later.

Reply:
Yeah I think the binary encoding can be dead simple: the number of elements in the list, the oid of the elements, and then the binary encoding of each element. Just figured we could punt to another commit.